frame/fast-unstake/src/lib.rs

```diff
@@ -221,28 +234,23 @@ pub mod pallet {
 			let ctrl = ensure_signed(origin)?;
 
 			ensure!(ErasToCheckPerBlock::<T>::get() != 0, <Error<T>>::CallNotAllowed);
-
-			let ledger =
-				pallet_staking::Ledger::<T>::get(&ctrl).ok_or(Error::<T>::NotController)?;
-			ensure!(!Queue::<T>::contains_key(&ledger.stash), Error::<T>::AlreadyQueued);
+			let stash = T::Staking::stash_by_ctrl(&ctrl).map_err(|_| Error::<T>::NotController)?;
+			ensure!(!Queue::<T>::contains_key(&stash), Error::<T>::AlreadyQueued);
 			ensure!(
-				Head::<T>::get().map_or(true, |UnstakeRequest { stash, .. }| stash != ledger.stash),
+				Head::<T>::get().map_or(true, |UnstakeRequest { stash, .. }| stash != stash),
```

Comment:
isn't this shadowing the variable stash from the larger scope? 😱 if so, isn't there a compiler warning? why is no test failing? dangerous code. 

either way, please use a different name

Reply:
It's not shadowing somehow, I thought it's a bit confusing, but looks kinda great :D Yes, will change the name.

Reply:
would be curious to figure out exactly what is happening here, it seems important and I can see a similar mistake causing damage someday.

Reply:
It's not a mistake. It works. Because `stash` in this case is a pattern that we match. So it works fine. We can't say `UnstakeRequest { stashAccount .. }` for example, because the field matches exactly.
And this is passing tests. So technically this is not a mistake, but it's confusing. So I will change the outer variable name.

Reply:
Scratch that, this should actually be shadowing the outer scope. But it does not warn.

Reply:
Yep, here's the deal:
1. It does shadow, so the equality comparison would always true
2. The reason it does not warn in this case is simply because the shadowing is only happening within `map_or` closure, so there's no `unused_variable` warning, because we use the original variable downstream.

Et voila!